### PR TITLE
fortify scan implementation

### PR DESCRIPTION
**Pull request by kaanaktas**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CCD-1257

### Change description ###

Implementation of fodScanRepository

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: config/fortify-client.properties [hunk 1]
```diff
@@ -0,0 +1,3 @@
+fortify.client.releaseId=65430
+fortify.exclude.pattern=*.jar|*.zip|*.tar|/build|/.gradle|/.git|*.xml
+fortify.client.unacceptableSeverity=HIGH
```
Comment (MSancaktutar) [inline config/fortify-client.properties:3]: default should be Low, we shouldn't need to set it here.

File: config/fortify-client.properties [hunk 2]
```diff
@@ -0,0 +1,3 @@
+fortify.client.releaseId=65430
```
Comment (MSancaktutar) [inline config/fortify-client.properties:1]: only this line should stay.

File: build.gradle [hunk 3]
```diff
@@ -164,6 +164,7 @@ dependencies {
 
     testCompile group: 'com.github.hmcts', name: 'ccd-test-definitions', version: '7.0.0'
     testCompile group: 'com.github.hmcts', name: 'befta-fw', version: '8.2.0'
+    testCompile "com.github.hmcts:fortify-client:0.0.3.ccd-1253.1:all"
```
Comment (MSancaktutar) [inline build.gradle:167]: will need to be replaced with stable release 1.0.0, when ready.

**Review by MSancaktutar — CHANGES_REQUESTED** [review]
Only minor comments to be actioned please, otherwise all good.

File: Jenkinsfile_nightly [hunk 4]
```diff
@@ -29,6 +29,7 @@ static LinkedHashMap<String, Object> secret(String secretName, String envVar) {
 withNightlyPipeline(type, product, component) {
     loadVaultSecrets(secrets)
     enableMutationTest()
-    enableDbMigration()
+    enableFortifyScan()
+    enableDbMigration('ccd')
```
Comment (MSancaktutar) [inline Jenkinsfile_nightly:33]: let's raise a separate PR for this line and merge it, so that this PR contains only its own changes.

**Review by MSancaktutar — CHANGES_REQUESTED** [review]
Please also fit in the after() clause for Fortify Scan stage to publish the scan report on the build.

**Review by MSancaktutar — APPROVED** [review]
LGTM.
Library branch selection will obviously be removed before merging this. Many thanks.

**Review by MSancaktutar — APPROVED** [review]
LGTM - just please remove the branch tag from pipeline library.

**Comment by kaanaktas** [comment]
> LGTM.
> Library branch selection will obviously be removed before merging this. Many thanks.

jenkins library has been merged and the library branch tag has been removed.

**Pull request merged**